### PR TITLE
test(card-section-images): add e2e tests

### PR DESCRIPTION
**Pull request by BrunnoM7**
### Related Ticket(s)

[Test Scenario]: Card-section-images #7447

### Description

Created e2e test cases for the `card-section-images` component covering the scenarios:
- Check cards with images, content is loaded and whole card clickable
- Check heading (bold) left aligned.
- Check the content (Eyebrow, heading, copy and arrow link) is loading on the card
- Customize theme from Carbon theme and verify gray 10, gray 90 and gray 100 theme.

### Changelog

**New**

- e2e test cases for `card-section-images` in both react and web components

### CODE REVIEW

File: packages/react/tests/e2e-storybook/cypress/integration/CardSectionImages/CardSectionImages.e2e.js [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+        );
+      });
+
+    cy.screenshot();
+
+    cy.percySnapshot(
+      'dds-card-section-images | default (desktop) | should load cards content',
+      {
+        widths: [1280],
+      }
```
Comment (jeffchew) [inline packages/react/tests/e2e-storybook/cypress/integration/CardSectionImages/CardSectionImages.e2e.js:76]: Replace with `cy.takeSnapshots()`

File: packages/react/tests/e2e-storybook/cypress/integration/CardSectionImages/CardSectionImages.e2e.js [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    );
+  });
+
+  it('should load the g10 theme', () => {
```
Comment (jeffchew) [inline packages/react/tests/e2e-storybook/cypress/integration/CardSectionImages/CardSectionImages.e2e.js:80]: Replace all of the theme tests with:

```
cy.carbonThemesScreenshot(
      {},
      {
        widths: [320],
      }
    );
```

File: packages/react/tests/e2e-storybook/cypress/integration/CardSectionImages/CardSectionImages.e2e.js [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+    cy.percySnapshot(
+      'dds-card-section-images | default (mobile) | should load cards content',
+      {
+        widths: [320],
+      }
+    );
+  });
+
+  it('should load the g10 theme', () => {
```
Comment (jeffchew) [inline packages/react/tests/e2e-storybook/cypress/integration/CardSectionImages/CardSectionImages.e2e.js:208]: cy.carbonThemesScreenshot(
      {},
      {
        widths: [320],
      }
    );

File: packages/react/tests/e2e-storybook/cypress/integration/CardSectionImages/CardSectionImages.e2e.js [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+      });
+
+    cy.screenshot();
+
+    cy.percySnapshot(
+      'dds-card-section-images | default (mobile) | should load cards content',
+      {
+        widths: [320],
+      }
+    );
```
Comment (jeffchew) [inline packages/react/tests/e2e-storybook/cypress/integration/CardSectionImages/CardSectionImages.e2e.js:205]: Replace with `cy.takeSnapshots('mobile')`

File: packages/web-components/tests/e2e-storybook/cypress/integration/card-section-images/card-section-images.e2e.js [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+      .find('svg path')
+      .each($icon => {
+        expect($icon).to.have.attr('d', 'M11.8 2.8L10.8 3.8 16.2 9.3 1 9.3 1 10.7 16.2 10.7 10.8 16.2 11.8 17.2 19 10z');
+      });
+
+    cy.screenshot();
+
+    cy.percySnapshot('dds-card-section-images | default (desktop) | should load cards content', {
+      widths: [1280],
+    });
```
Comment (jeffchew) [inline packages/web-components/tests/e2e-storybook/cypress/integration/card-section-images/card-section-images.e2e.js:70]: Replace with `cy.takeSnapshots()`

File: packages/web-components/tests/e2e-storybook/cypress/integration/card-section-images/card-section-images.e2e.js [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+  });
+
+  it('should load the g10 theme', () => {
```
Comment (jeffchew) [inline packages/web-components/tests/e2e-storybook/cypress/integration/card-section-images/card-section-images.e2e.js:73]: Replace all of the theme tests with:

```
it('should load correctly in all themes', () => {
    cy.visit(`/${_pathDefault}`);
    cy.viewport(1280, 780);

    cy.carbonThemesScreenshot();
  });
```

File: packages/web-components/tests/e2e-storybook/cypress/integration/card-section-images/card-section-images.e2e.js [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+      .find('svg path')
+      .each($icon => {
+        expect($icon).to.have.attr('d', 'M11.8 2.8L10.8 3.8 16.2 9.3 1 9.3 1 10.7 16.2 10.7 10.8 16.2 11.8 17.2 19 10z');
+      });
+
+    cy.screenshot();
+
+    cy.percySnapshot('dds-card-section-images | default (mobile) | should load cards content', {
+      widths: [320],
+    });
```
Comment (jeffchew) [inline packages/web-components/tests/e2e-storybook/cypress/integration/card-section-images/card-section-images.e2e.js:173]: Replace with `cy.takeSnapshots('mobile')`

File: packages/web-components/tests/e2e-storybook/cypress/integration/card-section-images/card-section-images.e2e.js [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+  });
+
+  it('should load the g10 theme', () => {
```
Comment (jeffchew) [inline packages/web-components/tests/e2e-storybook/cypress/integration/card-section-images/card-section-images.e2e.js:176]: Replace all of the theme tests with:

```
it('should load correctly in all themes', () => {
    cy.visit(`/${_pathDefault}`);
    cy.viewport(320, 780);

   cy.carbonThemesScreenshot(
      {},
      {
        widths: [320],
      }
    );

  });
```

**Review by jeffchew — CHANGES_REQUESTED** [review]
Hey @BrunnoM7 see my comments, similar to the other e2e test PR, a lot of the Percy snapshot lines can be replaced with the global methods now.

**Comment by BrunnoM7** [comment]
Thanks for reviewing @jeffchew. I just applied the changes you mention.

**Review by jeffchew — APPROVED** [review]
Thank you for making the changes!

**Pull request merged**